synapse/dendrite/coordinator.py

```diff
@@ -0,0 +1,277 @@
+        queue names for a given MIME type. This method is used to find all the queue names that are associated with a
+        specified MIME type.
+
+        Args:
+            mimeType (str):
+                The MIME type as is returned by libmagic.
+
+        Returns:
+            list: A list of queue names, or an empty list if none are found.
+        '''
+        queues = []
+        for tufo in self.core.getTufosByProp('dendrite:coordinator:mime', mimeType):
+            queues.append(tufo[1].get('dendrite:coordinator:queue'))
+        return queues
+
+    def _guid(self, mimeType, queue):
+        return s_common.guid((mimeType, queue))
+
+    def _registration(self, mimeType, queue):
+        return self.core.getTufoByProp('dendrite:coordinator', self._guid(mimeType, queue))
+
+    def _registerListeners(self):
+        self.listenCore.on('tufo:add:file:bytes', self._fileAdded)
+        self.listenCore.on('tufo:props:file:bytes', self._fileUpdated)
+
+    def _fileAdded(self, mesg):
+        tufo = mesg[1].get('tufo')
+        logger.debug('Received file added event: %s', tufo)
+        self._processTufo(tufo)
+
+    def _fileUpdated(self, mesg):
+        tufo = mesg[1].get('tufo')
+        logger.debug('Received file updated event: %s', tufo)
+        self._processTufo(tufo)
+
+    def _processTufo(self, tufo):
+        self._updateState()
+        guid = tufo[1].get('file:bytes')
+        blob = self.axon.byiden(guid)
+        blob[1]['axon:blob:size'] = min(Coordinator._BYTES_TO_READ, tufo[1].get('file:bytes:size'))
+        bytes = b''.join(self.axon.iterblob(blob))
+        mimeType = self._getMimeType(bytes)
+        logger.debug('Detected mime: %s for guid: %s', mimeType, guid)
+        job = {'axon': self.axonLink, 'guid': guid, 'mime': mimeType}
+        for queue in self.queues(mimeType):
+            logger.debug('Adding job:put in queue %s', queue)
+            self.jobs.put(queue, job)
+
+    def _getMimeType(self, bytes):
+        mimeType = None
+        try:
+            mimer = self._createMimer()
+            mimeType = mimer.from_buffer(bytes)
+        except magic.MagicException as e:
+            logger.error('Error determining mime type: %s', e)
+        return mimeType
+
+    def _createMimer(self):
+        if self.magicFile:
+            return magic.Magic(mime=True, magic_file=self.magicFile)
+        else:
+            return magic.Magic(mime=True)
+
+    def _assertOpts(self):
+        if not Coordinator._REQUIRED_ARGS.issubset(self.opts):
+            raise TypeError("kwargs missing required args: %s" % (Coordinator._REQUIRED_ARGS - set(self.opts.keys())))
+
+    def _assertConnected(self):
+        try:
+            self.listenCore.ping()
+            logger.info('Successfully connected to cortex')
+        except s_exc.NoSuchObj:
+            raise s_exc.NoSuchObj(msg='unable to connect to cortex, please check url')
```

Comment:
Though SynErr args are totally free-form, we're kinda trying to stick to some conventions for them.  I think the others have "mesg=" rather than "msg="  ( might also be nice to plumb through the rest of the SynErr props in general )